src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -452,6 +455,11 @@ public void saveProgram(File f) throws IOException {
             PROGRAM_STATE_SERIALIZATION.set(old);
             Util.deleteFile(tmpFile);
         }
+
+        if (logging) {
+            long end = System.nanoTime();
+            LOGGER.log(FINER, "saved {0} of size {1}Kb in {2}ms", new Object[] {f, f.length() / 1000, (end - start) / 1000 / 1000});
```

Comment:
Also, logging the I/O size and time :champagne: 

IIRC we can set up logging for just this class to FINER level, and get the cost of that I/O operation right?